### DISCUSSION TITLE
Migrating from elasticsearch to meilisearch

### DISCUSSION THREAD

**Discussion by valentinhttr** [meilisearch/product#697] [Feedback & Feature Proposal]
Hi,

I'm evaluating the possibilities of migrating from Elasticsearch to Meilisearch because I have a very specific need and Elasticsearch is a bit of a gas factory for the only need I have.

I need to be able to search PDF and DOCX documents. I know it's not possible to do this directly with Meilisearch, but I already have a script that allows me to extract the text, so I could simply index the text with the corresponding model id to meilisearch.

My question is whether it's possible to do text search according to this need:
1. Search by one or more **exact** words - case sensitive (e.g. adding quotes ("") around the word).
2. Search for one or more **partial** words - case insensitive? (ex: `ell` matches `hello`)

I could create my own little search "language" on top of meilisearch if that's possible. For example, I need to be able to do a search like: `"heLLo" AND ben or jame`.

Is this possible?
Thanks

**Comment by macraig** [comment]
Hi @valh1996 , apologies for the late reply!

> Search by one or more exact words - case sensitive (e.g. adding quotes ("") around the word).

Yes, you can use quotes for exact word matching

> Search for one or more partial words - case insensitive? (ex: ell matches hello)

Meilisearch can handle prefixes (see [prefix search](https://www.meilisearch.com/docs/learn/advanced/prefix), but for the particular example you gave it won't work as `ell` has a typo in the first character, and the engine won't accept typos for words with less than 5 characters. You can customize the char minimum (see the [typos](https://www.meilisearch.com/docs/learn/configuration/typo_tolerance) in the docs) but the other factor here is that a typo in the first character is counted as two typos.
